backend/controller/dal/async_calls.go

```diff
@@ -157,17 +170,44 @@ func (d *DAL) CompleteAsyncCall(ctx context.Context, call *AsyncCall, result eit
 				ScheduledAt:       time.Now().Add(call.Backoff),
 			})
 			if err != nil {
-				return dalerrs.TranslatePGError(err)
+				return false, dalerrs.TranslatePGError(err) //nolint:wrapcheck
+			}
+			isFinalResult = false
+			didScheduleAnotherCall = true
+		} else if call.RemainingAttempts == 0 && call.CatchVerb.Ok() {
+			// original error is the last error that occurred before we started to catch
+			originalError := call.Error.Default(result.Get())
+			//scheduledAt should be immediate if this is our first catch attempt, otherwise we should use backoff
+			scheduledAt := time.Now()
+			if call.Catching {
+				scheduledAt = scheduledAt.Add(call.Backoff)
 			}
+			_, err = d.db.FailAsyncCallWithRetry(ctx, sql.FailAsyncCallWithRetryParams{
+				ID:                call.ID,
+				Error:             result.Get(),
+				RemainingAttempts: 0,
+				Backoff:           call.Backoff, // maintain backoff
+				MaxBackoff:        call.MaxBackoff,
+				ScheduledAt:       scheduledAt,
+				Catching:          true,
+				OriginalError:     optional.Some(originalError),
+			})
+			if err != nil {
+				return false, dalerrs.TranslatePGError(err) //nolint:wrapcheck
+			}
+			isFinalResult = false
+			didScheduleAnotherCall = true
 		} else {
 			_, err = tx.db.FailAsyncCall(ctx, result.Get(), call.ID)
 			if err != nil {
-				return dalerrs.TranslatePGError(err)
+				return false, dalerrs.TranslatePGError(err) //nolint:wrapcheck
 			}
 		}
 	}
-
-	return finalise(tx)
+	if err := finalise(tx, isFinalResult); err != nil {
+		return false, err
```

Comment:
Just confirming to make sure I understand correctly - we aren't returning `didScheduleAnotherCall, err` here because if `finalise` returned an error, then the enqueuing of the retry or catch call in this same `tx` would have failed, thus we did not actually successfully schedule another call?

Reply:
that's right, ther whole tx is cancelled

Reply:
sweet thanks :)